target/linux/ramips/base-files/etc/board.d/02_network

```diff
@@ -337,6 +338,10 @@ ramips_setup_macs()
 		lan_mac=$(mtd_get_mac_binary factory 40)
 		wan_mac=$(mtd_get_mac_binary factory 46)
 		;;
+	rb750gr3)
+		wan_mac=$(cat /sys/class/net/eth0/address)
+		lan_mac=$(macaddr_add "$wan_mac" 1)
```

Comment:
mtd-mac-address-increment shouldn't care about the switch vlan layout. The change should be applied to all vlans and you need to fix it for the wan vlan afterwards. But I'm not entirely sure since I don't have hardware to test.

Please give it a try and report back.